solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -160,6 +195,54 @@ fn compute_uniswap_in(out: U256, shortage: &TokenContext, excess: &TokenContext)
     U256::from(1000) * out * excess.reserve / (U256::from(997) * (shortage.reserve - out)) + 1
 }
 
+///
+/// Returns true if for each trade the executed price is not smaller than the limit price
+/// Thus we ensure that `buy_token_price / sell_token_price >= limit_buy_amount / limit_sell_amount`
+///
+fn is_valid_solution(solution: &SinglePairSettlement) -> bool {
+    for trade in solution.trades.iter() {
+        let order = trade.order;
+        let buy_token_price = match solution.clearing_prices.get(&order.buy_token) {
+            Some(price) => price,
+            None => return false,
```

Comment:
Shouldn't this panic because it should be guaranteed at this point that every token has a price?